### PR TITLE
fix: Remove deprecations for 1.10.0

### PR DESCRIPTION
**Pull request by spydon**
# Description
Removed deprecations for 1.10.0 and fixes some small unreleased regressions found when going through the examples.

## Checklist

- [x] I have followed the [Contributor Guide] when preparing my PR.
- [x] I have updated/added tests for ALL new/updated/fixed functionality.
- [x] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [x] I have updated/added relevant examples in `examples` or `docs`.

## Breaking Change?

- [x] No, this PR is not a breaking change.

## Related Issues

[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Conventional Commit]: https://conventionalcommits.org
[CHANGELOG]: https://github.com/flame-engine/flame/blob/main/CHANGELOG.md

### CODE REVIEW

File: doc/flame/camera_component.md [hunk 1]
```diff
@@ -262,8 +262,7 @@ if (!camera.canSee(component)) {
 
 ## Comparison to the deprecated camera
```
Comment (luanpotter) [inline doc/flame/camera_component.md:263]: can we remove the entire section now?

File: doc/flame/inputs/tap_events.md [hunk 2]
```diff
@@ -167,54 +167,50 @@ class MyComponent extends Component with TapCallbacks {
...
+
+- The argument pair such as `(int pointerId, TapDownDetails details)` was replaced with a single
+  event object `TapDownEvent event`.
+- There is no return value anymore, but if you need to make a component to pass-through the taps
+  to the components below, then set `event.continuePropagation` to true. This is only needed for
+  `onTapDown` events -- all other events will pass-through automatically.
+- If your component needs to know the coordinates of the point of touch, use
+  `event.localPosition` instead of computing it manually. Properties `event.canvasPosition` and
+  `event.devicePosition` are also available.
+- If the component is a `PositionComponent`, then make sure its size is set correctly (for
```
Comment (luanpotter) [inline doc/flame/inputs/tap_events.md:212]: I understand what you mean with `for example by turning on the debug mode`, but I think it is a bit misleading, seem to be implying turning that on will cause the size to be set.

File: packages/flame/lib/src/camera/viewfinder.dart [hunk 3]
```diff
@@ -19,21 +18,17 @@ import 'package:meta/meta.dart';
 /// If you add children to the [Viewfinder] they will appear like HUDs i.e.
 /// statically in front of the world.
 class Viewfinder extends Component
```
Comment (luanpotter) [inline packages/flame/lib/src/camera/viewfinder.dart:20]: why no longer parent is a camera?

File: packages/flame/lib/src/camera/viewfinder.dart [hunk 4]
```diff
@@ -83,7 +78,7 @@ class Viewfinder extends Component
   }
 
   /// Reference to the parent camera.
-  CameraComponent get camera => parent;
+  CameraComponent get camera => parent! as CameraComponent;
```
Comment (luanpotter) [inline packages/flame/lib/src/camera/viewfinder.dart:81]: why is this better?

File: packages/flame/lib/src/camera/viewfinder.dart [hunk 5]
```diff
@@ -19,21 +18,17 @@ import 'package:meta/meta.dart';
...
 class Viewfinder extends Component
-    with ParentIsA<CameraComponent>
     implements AnchorProvider, AngleProvider, PositionProvider, ScaleProvider {
-  /// Internal transform matrix used by the viewfinder.
-  final Transform2D _transform = Transform2D();
-
-  @internal
-  Transform2D get transform => _transform;
+  /// Transform matrix used by the viewfinder.
+  final Transform2D transform = Transform2D();
```
Comment (luanpotter) [inline packages/flame/lib/src/camera/viewfinder.dart:23]: I guess we can't prevent people from doing `viewport.transform.transformMatrix.storage[0] = 0` or something...

File: packages/flame/lib/src/experimental/fixed_integer_resolution_viewport.dart [hunk 6]
```diff
@@ -1,97 +0,0 @@
...
-
-/// This viewport is a very similar to [FixedResolutionViewport], but allows
-/// for better handling of viewing pixel art. The main point is it ensures
-/// sprites align to the physical pixel grid and only scales at integer
-/// intervals. This prevents artifacts or distortion happening to scaled up
-/// pixel art as discussed in https://github.com/flame-engine/flame/issues/1152.
-/// Remember to set [devicePixelRatio] before using the viewport; a good place
-/// to update it is in handleResize of your game, directly from
-/// WidgetBindings.instance.window.devicePixelRatio.
-class FixedIntegerResolutionViewport extends Viewport {
```
Comment (luanpotter) [inline packages/flame/lib/src/experimental/fixed_integer_resolution_viewport.dart:14]: why remove?

File: packages/flame/lib/src/game/flame_game.dart [hunk 7]
```diff
@@ -220,12 +201,6 @@ class FlameGame<W extends World> extends ComponentTreeRoot
         Duration.microsecondsPerSecond;
   }
 
-  @override
-  Projector get viewportProjector => oldCamera.viewport;
```
Comment (luanpotter) [inline packages/flame/lib/src/game/flame_game.dart:224]: do still have anything that uses this interface Projector?

File: packages/flame/test/components/mixins/component_viewport_margin_test.dart [hunk 8]
```diff
@@ -41,4 +41,4 @@ void main() {
 }
 
 class _ComponentWithViewportMargin extends PositionComponent
-    with HasGameRef, ComponentViewportMargin {}
+    with HasGameReference, ComponentViewportMargin {}
```
Comment (luanpotter) [inline packages/flame/test/components/mixins/component_viewport_margin_test.dart:44]: did we reach a consensus about this? should this be a separate PR/discussion?

**Review by luanpotter — APPROVED** [review]
LGTM, thank you so much for finally deleting the old code path! That makes me so much happy...

File: packages/flame/lib/src/camera/viewfinder.dart (hunk 3 above)
Comment (spydon) [reply packages/flame/lib/src/camera/viewfinder.dart:20]: Initially it wasn't there, then I added it in the last PR, but some operations need to check whether it has a parent or not and `parentIsA` does a `!` on the parent, so we couldn't do those checks anymore which led to some bugs.

File: doc/flame/inputs/tap_events.md (hunk 2 above)
Comment (spydon) [reply doc/flame/inputs/tap_events.md:212]: I didn't write this, but I can fix it up!

File: packages/flame/lib/src/camera/viewfinder.dart (hunk 4 above)
Comment (spydon) [reply packages/flame/lib/src/camera/viewfinder.dart:81]: It's not, but we have to be able to check if `parent` is null at some places and `parentIsA` doesn't allow that.

File: packages/flame/lib/src/camera/viewfinder.dart (hunk 5 above)
Comment (spydon) [reply packages/flame/lib/src/camera/viewfinder.dart:23]: Yeah, if they do that they'll have to suffer the consequences 😂

File: packages/flame/lib/src/experimental/fixed_integer_resolution_viewport.dart (hunk 6 above)
Comment (spydon) [reply packages/flame/lib/src/experimental/fixed_integer_resolution_viewport.dart:14]: Because it's a viewport for the old camera

File: packages/flame/lib/src/game/flame_game.dart (hunk 7 above)
Comment (spydon) [reply packages/flame/lib/src/game/flame_game.dart:224]: Not after this PR

File: doc/flame/inputs/tap_events.md (hunk 2 above)
Comment (spydon) [reply doc/flame/inputs/tap_events.md:212]: I realized this point is not even related to the migration so I removed it.

File: packages/flame/lib/src/experimental/fixed_integer_resolution_viewport.dart (hunk 6 above)
Comment (spydon) [reply packages/flame/lib/src/experimental/fixed_integer_resolution_viewport.dart:14]: Created #2810 to track this for the new camera, since it was in the experimental section I don't think it matters much that we remove it momentarily (it was used for a specific rendering bug in Flutter).
